### PR TITLE
Add missing dependencies

### PR DESCRIPTION
**Pull request by dmentipl**
Addresses #1435 

Add missing dependencies to `setup.cfg`. These are packages that are imported directly in IMPROVER code but were not listed as dependencies.

This PR also updates the Python 3.7 / Iris 2.4 Conda environment file. This includes removing pip and packages installed with pip from PyPI as all packages are now available from conda-forge.

There is also a PR for the conda-forge feedstock adding these changes there too: see https://github.com/conda-forge/improver-feedstock/pull/2

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -19,12 +19,18 @@ setup_requires =
...
+    cartopy
     cftime == 1.0.1
+    cf_units
     clize
+    dask
+    netCDF4
     numpy
-    sigtools
+    python-dateutil
+    pytz
```
Comment (lucyleeow) [inline setup.cfg:30]: I wonder if this can be optional since it seems to only be used for generating ancillary files. (no worries if not)

**Review by lucyleeow — COMMENTED** [review]
LGTM, thanks for separating out the dependencies into groups, this is nicer (should we do this for py36_iris22 as well?)

**Review by lucyleeow — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (lucyleeow) [reply setup.cfg:30]: Also why is the version pinned in the environment yaml file but not here..?

**Comment by dmentipl** [comment]
> should we do this for environment_py36_iris22 as well?

I was concerned with breaking things for others so I left that environment untouched. In addition, for some unknown reason, version 4.2.0 of timezonefinder is unavailable from conda-forge. So removing pip is not possible for that environment file. However, the other pip installed dependencies *are* available on conda-forge now.

Note also that (a specific version of) proj is specified as a dependency in that environment file. It is not directly imported but is a dependency of cartopy (see #1424).

In summary, I would be happy to reorganise and add dependencies to that environment file but only if Met Office people carefully checked it.

File: setup.cfg (hunk 1 above)
Comment (dmentipl) [reply setup.cfg:30]: My thinking was that any package directly imported in IMPROVER should be included. When I say "directly imported" I mean without using, e.g., importlib as is the case for numba and the other optional dependencies.

Comment (dmentipl) [reply setup.cfg:30]: I've unpinned it now in the Python 3.7 environment file.

**Review by lucyleeow — APPROVED** [review]

**Review by tjtg — COMMENTED** [review]
The text changes to `environment_py37_iris24.yml` and `setup.cfg` all look ok. Using comments in the YAML to separate required, optional and development helps clarify the intention.

However, when I created a fresh conda environment using `environment_py37_iris24.yml`, I get acceptance test failures from all tests in `test_generate_timezone_mask_ancillary.py`. My first suspect here was the pytz package as it includes a copy of the Olson tz database, so any timezone change that goes into the database will result in a different output. Setting pytz back to the pinned version 2020.5 significantly reduced the number of mismatched elements (eg. 24 -> 2) reported by these tests, but didn't change any of them to passing. Since the timezones are integer data, the numpy allclose relative and absolute tolerances don't give any allowance for differences. The number of mismatched elements is very small (eg. 2 out of 300,000).

It looks like there's currently some dependence on the timezonefinder version.
Another option would be to change the comparison function and add tolerance for a small number of mismatched elements.

**Comment by dmentipl** [comment]
Thanks for the review @tjtg. I've pinned both pytz and timezonefinder. Now the acceptance tests pass. I've also made `setup.cfg` and `environment_py37_iris24.yml` consistent with each other.

**Review by tjtg — APPROVED** [review]
